### PR TITLE
Sssp with traceback

### PR DESCRIPTION
**Pull request by tommytgraph**
SSSP with traceback. Write_size parameter allows users to trade performance for space and vice versa.

### CODE REVIEW

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql [hunk 1]
```diff
@@ -0,0 +1,156 @@
+CREATE QUERY tg_shortest_ss_pos_wt_tb (VERTEX source, SET<STRING> v_type, SET<STRING> e_type,
+ STRING wt_attr, STRING wt_type, FLOAT epsilon = 0.001,BOOL print_accum = TRUE, INT output_limit = -1, 
+ BOOL display_edges = FALSE, STRING result_attr = "", BOOL spit_to_file = FALSE,
```
Comment (wyatt-joyner-tg) [inline algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:3]: Remove `spit_to_file` if unused

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  component = {v_type};
+  component = component MINUS start_set; 
+   
+  component = SELECT s 
+              FROM component:s
+              WHERE s.@prevMinPath != -1
+              POST-ACCUM 
+			          IF result_attr != "" THEN s.setAttr(result_attr, s.@minPath.top().cost) END;
+
+  //store and write paths by batches to reduce memory consumption
```
Comment (wyatt-joyner-tg) [inline algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:94]: Are we actually using batches?

**Review by wyatt-joyner-tg — CHANGES_REQUESTED** [review]
Logic looks good. We'll stylistically update after the PR and rename some of the variables for clarity. I think we might be able to write the path formulation and printing a bit more clearly, as well.

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql (hunk 1 above)
Comment (wyatt-joyner-tg) [reply algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:3]: I think result_attr is also unused?

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql (hunk 2 above)
Comment (tommytgraph) [reply algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:94]: We only use batches for the traceback, which is often very memory consuming. For the other parts, the memory consumed = the local accumulator overhead.

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+   
+	##### Initialize #####
+	start = {source};
+    start = SELECT s 
+          FROM start:s
+          POST-ACCUM s.@minPath = min_tup(0, 1, s);
+  
+
+	##### Walk one step at a time until there is no more path.
+	WHILE start.size() != 0 DO
```
[the remark this answers is not in the document]
Comment (tommytgraph) [reply algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:50]: have a limit clause can lead to false result. I don't know if the performance is worth it

[the remark this answers is not in the document]
Comment (wyatt-joyner-tg) [reply algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:50]: That's a good point!

File: algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql (hunk 2 above)
Comment (wyatt-joyner-tg) [reply algorithms/Path/shortest_path/tg_shortest_ss_pos_wt_tb.gsql:94]: Ah, gotcha. Most batch implementations use a FOREACH with a mod of the batch size to randomly sample. I see component MINUS tmp now, though, haha.

**Review by wyatt-joyner-tg — APPROVED** [review]

**Pull request merged**